src/poetry/core/masonry/builders/sdist.py

```diff
@@ -320,17 +321,21 @@ def find_files_to_add(self, exclude_build: bool = False) -> set[BuildIncludeFile
         to_add = super().find_files_to_add(exclude_build)
 
         # add any additional files, starting with all LICENSE files
-        additional_files = set(self._path.glob("LICENSE*"))
+        additional_files: set[str | Path] = set(self._path.glob("LICENSE*"))
```

Comment:
Seems like these should all be `Path` -- when can `glob` return a `str`? `convert_script_files` is in this file and can only return `Path`.

Reply:
See the other discussion, I need to also add `str`, since is what `poetry build` add to `local_config["readme"]`.

Of course I can convert to `Path` when I add to the set, it is an idea :bulb:

Reply:
Making sure I'm understandable.

Variable `additional_files` evaluates **initially** after call to `glob` as a `set[Path]`. Period.

But then, I also need to include to the `set` the readme file(s) from `local_config["readme"]` which is a `str` or a `list[str]`.

Reply:
Evidences of `local_config["readme"]` be a `str` or `list[str]`

https://github.com/python-poetry/poetry-core/blob/0759db61f01f008edc3ffe2b790ac9ecc892da77/src/poetry/core/factory.py#L48

https://github.com/python-poetry/poetry-core/blob/0759db61f01f008edc3ffe2b790ac9ecc892da77/src/poetry/core/pyproject/toml.py#L72-L74

Reply:
Although it's not necessary, I think it might be worth to just make sure that `additional_files` are always of type `Path` (for cleaner code and to avoid confusion). Proposal see below.

Reply:
I liked this way. I was trying to avoid extra work, but I completely agree that makes much cleaner.